assignment3.txt

```diff
@@ -0,0 +1,71 @@
+Jinoo Hong
+
+1.  Determine the complexity of the following code snippets: 
+
+    a.  
+    ``````C
+      int sum = 0;
+      for(i = 0; i < n; i++)
+              sum++;
+    ```
+O(n)
+    
+    b.  
+    ``` C
+    sum = 0;
+    for( i = 0; i < n; i++ )
+        sum++;
+    for( i = 0; i < n; i++ )
+        for( j = 0; j < n; j++ )
+            sum++;
+    ```
+O(2n)
+
+    c.  
+    ``` C
+    for(i=0;i<n;i+=2) {
+  	  for(j=1;j<=n;j*=2) {
+  		  printf(“%d,%d\n”,i,j);
+	    }
+    }
+    ```
+O(n^2)
```

Comment:
Again, not quite:  The first for loop gives a time complexity of n since you iterate through n elements. The second for loop would normally also be n, but since it multiplies the j by two each time, that means you're eliminating half the number of elements each time - this causes a time complexity of log (n). And since they're nested, you multiply the two with each other so that the time complexity is O(nlog(n)).